defold-rive/src/script_rive.cpp

```diff
@@ -257,6 +257,57 @@ namespace dmRive
     }
 
 
+    static int RiveComp_GetTextRun(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+
+        RiveComponent* component = 0;
+        dmScript::GetComponentFromLua(L, 1, dmRive::RIVE_MODEL_EXT, 0, (void**)&component, 0);
+
+        const char* name = luaL_checkstring(L, 2);
+        const char* nested_artboard_path = 0;
+
+        if (lua_isstring(L, 3))
+        {
+            nested_artboard_path = lua_tostring(L, 3);
+        }
+
+        const char* text_run = CompRiveGetTextRun(component, name, nested_artboard_path);
+
+        if (!text_run)
+        {
+            return DM_LUA_ERROR("The text-run '%s' could not be found, or the text is empty.", name);
+        }
+
+        lua_pushstring(L, text_run);
+        return 1;
+    }
+
+    static int RiveComp_SetTextRun(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+
+        RiveComponent* component = 0;
+        dmScript::GetComponentFromLua(L, 1, dmRive::RIVE_MODEL_EXT, 0, (void**)&component, 0);
+
+        const char* name = luaL_checkstring(L, 2);
+        const char* nested_artboard_path = 0;
+
+        if (lua_isstring(L, 4))
+        {
+            nested_artboard_path = lua_tostring(L, 4);
+        }
+
+        const char* text_run = luaL_checkstring(L, 3);
+
+        if (!CompRiveSetTextRun(component, name, text_run, nested_artboard_path))
+        {
+            return DM_LUA_ERROR("The text-run '%s' could not be found, or the text is empty.", name);
```

Comment:
Is this error message correct? What do we mean by "or the text is empty?"

Reply:
Yeah I don't know actually. I've changed it now, thanks!